bank/api/api.go

```diff
@@ -0,0 +1,246 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/gorilla/context"
+	"github.com/iliyanmotovski/bankv2/bank/domain"
+	"net/http"
+	"time"
+)
+
+func SignUpHandler(userStore domain.UserStore) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var request domain.UserRegistrationRequest
+		json.NewDecoder(r.Body).Decode(&request)
+		w.Header().Set("Content-Type", "application/json")
+
+		u, err := userStore.GenerateUser(request)
```

Comment:
You can simplify api usage of the store with:

```golang
u := userStore.RegisterUser(request); 
```

t.e GenerateUser to be removed.